src/comment_parser.rs

```diff
@@ -0,0 +1,379 @@
+//! Implements markup parsing for metamath theorem comments.
+//!
+//! The [set.mm](https://github.com/metamath/set.mm/) library contains many
+//! theorems with comments, and those comments contain markup in a special syntax
+//! implemented by [metamath.exe](https://github.com/metamath/metamath-exe).
+//! This module implements a SAX-style parser which yields events about the
+//! beginning and end of each syntax event.
```

Comment:
This shall not be limited to `set.mm`, right?

Actually, the Metamath book, p. 139., chapter 4.4.1, "Comments in the Metamath language", specifies this formatting. It shall be referenced here.

Reply:
Oh, if there is a spec it would be much better to have used that. (Although I suspect that metamath.exe doesn't implement it correctly, there are some bits of cursed knowledge in this parser...)